File: backend/src/main/java/com/paperai/model/entity/PaperVersion.java
package com.paperai.model.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 论文版本实体
 *
 * @author: ch
 * @date 2026年05月11日
 */
@Data
@TableName("paper_version")
public class PaperVersion {

    /** 版本ID */
    @TableId(type = IdType.AUTO)
    private Long id;

    /** 所属论文ID */
    private Long paperId;

    /** 版本号 */
    private Integer versionNo;

    /** 阶段: DRAFT/REVIEWED/POLISHED/FINAL/MANUAL_EDIT */
    private String stage;

    /** 版本摘要/日志 */
    private String summary;

    /** 论文全文(Markdown) */
    private String content;

    /** 字数统计 */
    private Integer wordCount;

    /** 编辑类型: MANUAL/AGENT/BATCH */
    private String editType;

    /** 详细修改说明 */
    private String changeSummary;

    /** 创建时间 */
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
}

File: backend-py/app/models/paper_version.py
from datetime import datetime
from sqlalchemy import String, Text, Integer, BigInteger, DateTime, func, Index
from sqlalchemy.orm import Mapped, mapped_column
from .base import Base


class PaperVersion(Base):
    __tablename__ = "paper_version"
    __table_args__ = (
        Index("ix_paper_version_paper_id", "paper_id"),
        Index("ix_paper_version_no", "paper_id", "version_no"),
    )

    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
    paper_id: Mapped[int] = mapped_column(BigInteger, nullable=False)
    version_no: Mapped[int] = mapped_column(Integer, nullable=False)
    stage: Mapped[str] = mapped_column(String(20), nullable=False, default="DRAFT")
    summary: Mapped[str | None] = mapped_column(String(500))
    content: Mapped[str | None] = mapped_column(Text)
    word_count: Mapped[int] = mapped_column(Integer, default=0)
    edit_type: Mapped[str] = mapped_column(String(20), default="MANUAL")
    change_summary: Mapped[str | None] = mapped_column(Text)
    created_at: Mapped[datetime] = mapped_column(
        DateTime, server_default=func.now(), nullable=False
    )
